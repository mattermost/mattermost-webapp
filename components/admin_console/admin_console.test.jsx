// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import AdminConsole from 'components/admin_console/admin_console';

describe('components/AdminConsole', () => {
    const baseProps = {
        config: {
            TestField: true,
        },
        license: {},
        match: {
            url: '',
        },
        roles: {
            channel_admin: 'test',
            channel_user: 'test',
            team_admin: 'test',
            team_user: 'test',
            system_admin: 'test',
            system_user: 'test',
        },
        showNavigationPrompt: false,
        isCurrentUserSystemAdmin: false,
        actions: {
            getConfig: jest.fn(),
            getEnvironmentConfig: jest.fn(),
            setNavigationBlocked: jest.fn(),
            confirmNavigation: jest.fn(),
            cancelNavigation: jest.fn(),
            loadRolesIfNeeded: jest.fn(),
            editRole: jest.fn(),
        },
    };

    test('should redirect to / when not system admin', () => {
        const props = {
            ...baseProps,
            isCurrentUserSystemAdmin: false,
        };
        const wrapper = shallow(
            <AdminConsole {...props}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should generate the routes', () => {
        const props = {
            ...baseProps,
            isCurrentUserSystemAdmin: true,
        };
        const wrapper = shallow(
            <AdminConsole {...props}/>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
