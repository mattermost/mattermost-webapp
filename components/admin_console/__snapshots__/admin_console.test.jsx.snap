// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AdminConsole should generate the routes 1`] = `
<div
  className="admin-console__wrapper"
  id="adminConsoleWrapper"
>
  <Connect(AnnouncementBarController) />
  <Connect(injectIntl(SystemNotice)) />
  <Connect(injectIntl(AdminSidebar))
    onFilterChange={[Function]}
  />
  <div
    className="admin-console"
  >
    <Highlight
      filter=""
    >
      <Switch>
        <Route
          key="about/license"
          path="/about/license"
          render={[Function]}
        />
        <Route
          key="reporting/system_analytics"
          path="/reporting/system_analytics"
          render={[Function]}
        />
        <Route
          key="reporting/team_statistics"
          path="/reporting/team_statistics"
          render={[Function]}
        />
        <Route
          key="reporting/server_logs"
          path="/reporting/server_logs"
          render={[Function]}
        />
        <Route
          key="user_management/user/:user_id"
          path="/user_management/user/:user_id"
          render={[Function]}
        />
        <Route
          key="user_management/users"
          path="/user_management/users"
          render={[Function]}
        />
        <Route
          key="environment/web_server"
          path="/environment/web_server"
          render={[Function]}
        />
        <Route
          key="environment/database"
          path="/environment/database"
          render={[Function]}
        />
        <Route
          key="environment/file_storage"
          path="/environment/file_storage"
          render={[Function]}
        />
        <Route
          key="environment/image_proxy"
          path="/environment/image_proxy"
          render={[Function]}
        />
        <Route
          key="environment/smtp"
          path="/environment/smtp"
          render={[Function]}
        />
        <Route
          key="environment/push_notification_server"
          path="/environment/push_notification_server"
          render={[Function]}
        />
        <Route
          key="environment/rate_limiting"
          path="/environment/rate_limiting"
          render={[Function]}
        />
        <Route
          key="environment/logging"
          path="/environment/logging"
          render={[Function]}
        />
        <Route
          key="environment/session_lengths"
          path="/environment/session_lengths"
          render={[Function]}
        />
        <Route
          key="environment/developer"
          path="/environment/developer"
          render={[Function]}
        />
        <Route
          key="site_config/customization"
          path="/site_config/customization"
          render={[Function]}
        />
        <Route
          key="site_config/localization"
          path="/site_config/localization"
          render={[Function]}
        />
        <Route
          key="site_config/users_and_teams"
          path="/site_config/users_and_teams"
          render={[Function]}
        />
        <Route
          key="environment/notifications"
          path="/environment/notifications"
          render={[Function]}
        />
        <Route
          key="site_config/emoji"
          path="/site_config/emoji"
          render={[Function]}
        />
        <Route
          key="site_config/posts"
          path="/site_config/posts"
          render={[Function]}
        />
        <Route
          key="site_config/file_sharing_downloads"
          path="/site_config/file_sharing_downloads"
          render={[Function]}
        />
        <Route
          key="site_config/public_links"
          path="/site_config/public_links"
          render={[Function]}
        />
        <Route
          key="authentication/signup"
          path="/authentication/signup"
          render={[Function]}
        />
        <Route
          key="authentication/email"
          path="/authentication/email"
          render={[Function]}
        />
        <Route
          key="authentication/password"
          path="/authentication/password"
          render={[Function]}
        />
        <Route
          key="authentication/mfa"
          path="/authentication/mfa"
          render={[Function]}
        />
        <Route
          key="authentication/gitlab"
          path="/authentication/gitlab"
          render={[Function]}
        />
        <Route
          key="plugins/plugin_management"
          path="/plugins/plugin_management"
          render={[Function]}
        />
        <Route
          key="plugins/plugin_:plugin_id"
          path="/plugins/plugin_:plugin_id"
          render={[Function]}
        />
        <Route
          key="integrations/integration_management"
          path="/integrations/integration_management"
          render={[Function]}
        />
        <Route
          key="integrations/bot_accounts"
          path="/integrations/bot_accounts"
          render={[Function]}
        />
        <Route
          key="integrations/gif"
          path="/integrations/gif"
          render={[Function]}
        />
        <Route
          key="integrations/cors"
          path="/integrations/cors"
          render={[Function]}
        />
        <Route
          key="experimental/features"
          path="/experimental/features"
          render={[Function]}
        />
        <Redirect
          to="/about/license"
        />
      </Switch>
    </Highlight>
  </div>
  <DiscardChangesModal
    onCancel={[MockFunction]}
    onConfirm={[MockFunction]}
    show={false}
  />
  <Connect(ModalController) />
</div>
`;

exports[`components/AdminConsole should redirect to / when not system admin 1`] = `
<Redirect
  to="/"
/>
`;
